### PR TITLE
Add Test Navigator and Pagination to the Test Runner Page

### PR DESCRIPTION
**Pull request by isaacdurazo**
- Adding Test Navigator markup and styling to Test Run page
- Add bootstrap pagination to test run page
- Track when changes are dirty to be able to distinguish "skipped" tests from "in progress" tests

:exclamation: I spent a day trying to wire up these new buttons and the buttons in the top nav to be able to save partial results. I attempted to do this in the `TestIframe` on `componentWillUnmount` and via the `react-router` but I ran into issues with both approaches. I found a bug already in `main` that provides the warning "Warning: Can't call setState (or forceUpdate) on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method." It seems like by the time we are trying to save the state is already gone. Fixing the saving partial results will need to be done in a follow up PR.

:notebook: Improving the styling of the bottom pagination is also incomplete because @isaacdurazo is planning on rethinking this entire section in a follow up ticket.

### CODE REVIEW

File: client/components/TestRun/index.jsx [hunk 1]
```diff
@@ -192,14 +209,97 @@ class TestRun extends Component {
...
+                                    onClick={this.toggleTestNavigator}
+                                    className="test-navigator-toggle"
+                                >
+                                    Show Test Navigator
+                                </button>
+                            )}
+                        </aside>
+                        <button
+                            onClick={this.toggleTestNavigator}
+                            className="test-navigator-toggle show"
```
Comment (rmeritz) [inline client/components/TestRun/index.jsx:292]: @isaacdurazo - The way I did "show" looks weird. I cannot tell if it is the `css` or my implentation.

File: client/components/TestRun/index.jsx [hunk 2]
```diff
@@ -192,14 +209,97 @@ class TestRun extends Component {
...
+                                                ) {
+                                                    resultClassName =
+                                                        'complete';
+                                                }
+                                            } else if (
+                                                this.state.currentTestIndex -
+                                                    1 >
+                                                i
+                                            ) {
+                                                resultClassName = 'skipped';
```
Comment (rmeritz) [inline client/components/TestRun/index.jsx:258]: @isaacdurazo - I'm not sure how we should define what "skipped" means. I wonder if it should show up separately at all. 

I took a stab at it but we should discuss.

**Comment by rmeritz** [comment]
@isaacdurazo - I'm not sure how the behavior of this and the new pagination buttons are supposed to interact with the "Partial Results" work @evmiguel has been working on. Let's discuss this.

File: client/components/TestRun/index.jsx (hunk 2 above)
Comment (rmeritz) [reply client/components/TestRun/index.jsx:258]: Remove skipped.

File: client/components/TestRun/index.jsx [hunk 3]
```diff
@@ -192,14 +209,98 @@ class TestRun extends Component {
                 </Helmet>
                 <Container fluid>
                     <Row>
-                        <Col>{heading}</Col>
+                        <aside className="col-md-3 test-navigator">
```
Comment (rmeritz) [inline client/components/TestRun/index.jsx:212]: @spectranaut - Do you know the best way with `aria` to markup the table of contents?

I think we might need something for the `aside` and/or `ol` I definitely want to add a way to represent the result state for each `li` which is currently on in the icon.  

Do if you don't "know" is there a recommended way you would look something like this up?

**Comment by rmeritz** [comment]
This is now done except for the open questions on the PR and implementing auto-saving partial results when you navigate to another PR which depends on the work @evmiguel is doing now.

File: client/components/TestRun/TestRun.css [hunk 4]
```diff
@@ -1,13 +1,166 @@
...
+    width: 2px;
+    height: 15px;
+    left: 4px;
+    top: 2px;
+    background: white;
+}
+
+.show {
+    width: 2em;
+}
```
Comment (spectranaut) [inline client/components/TestRun/TestRun.css:68]: hey @isaacdurazo , do you know why this is here? it's causing the toast to look like this: 
problem-with-toast

**Review by spectranaut — APPROVED** [review]
Added commit to fix a few things

**Pull request merged**